sdk/log/crate/src/logger.rs

```diff
@@ -250,6 +267,10 @@ macro_rules! impl_log_for_unsigned_integer {
 
                             #[cfg(target_os = "solana")]
                             {
+                                // JC: Should we do a check on the size so that
+                                // the syscall isn't always used? For example, if
+                                // `written` is less than 32 bytes, then we do a
+                                // normal copy
```

Comment:
Spoke to Lucas about this. The compiler will always use the syscall if the size cannot be determined at compile time, so in this case the best we can do is to call the syscall.